src/app/pen.cpp

```diff
@@ -38,12 +39,19 @@ auto pen::process_events() -> event_loop_status
 
             // Move cursor to pen position, generate a click if the pen is
             // touching the screen
-            this->send_button_press(
-                screen_x, screen_y,
-                device_state.pressure > 0
-                    ? MouseButton::Left
-                    : MouseButton::None
-            );
+            MouseButton new_state =
+               device_state.pressure > 0
+                ? MouseButton::Left
+                : MouseButton::None;
+            this->send_button_press(screen_x, screen_y, new_state);
+
+            if (this->state != new_state) {
+                if (this->state == MouseButton::Left)
+                    this->screen.set_repainting_mode(repainting_mode::fast);
+                else
+                    this->screen.set_repainting_mode(repainting_mode::standard);
+            }
+            this->state = new_state;
```

Comment:
Can we simplify this to:

```suggestion
            if (new_state == MouseButton::Left)
            {
                this->screen.set_repainting_mode(repainting_mode::fast);
            }
            else
            {
                this->screen.set_repainting_mode(repainting_mode::standard);
            }
```

Which eliminates the need for the `state` member variable? Or did I miss something?

Reply:
Yes, you are right I forgot something: when you release the pen you want the screen to be redrawn at that moment to see your whole stroke (at least in my opinion). I have added this in the patch below.

Reply:
Makes sense, thanks!